integration-tests/suites/udp_networkflow.go

```diff
@@ -0,0 +1,293 @@
+				s.runTest(image, recv, send, port)
+			})
+
+			port++
+		}
+	}
+}
+
+func (s *UdpNetworkFlow) runTest(image, recv, send string, port uint16) {
+	server := s.runServer(config.ContainerStartConfig{
+		Name:    UDP_SERVER,
+		Image:   image,
+		Command: newServerCmd(recv, port),
+	}, port)
+	client := s.runClient(config.ContainerStartConfig{
+		Name:       UDP_CLIENT,
+		Image:      image,
+		Command:    newClientCmd(send, "", "", server),
+		Entrypoint: []string{"udp-client"},
+	})
+	log.Info("Server: %s - Client: %s\n", server.String(), client.String())
+
+	// Expected client connection
+	clientConnection := types.NetworkInfo{
+		LocalAddress:   "",
+		RemoteAddress:  fmt.Sprintf("%s:%d", server.ip, server.port),
+		Role:           "ROLE_CLIENT",
+		SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+		CloseTimestamp: types.NilTimestamp,
+	}
+
+	// Expected server connection
+	serverConnection := types.NetworkInfo{
+		LocalAddress:   fmt.Sprintf(":%d", server.port),
+		RemoteAddress:  client.ip,
+		Role:           "ROLE_SERVER",
+		SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+		CloseTimestamp: types.NilTimestamp,
+	}
+
+	s.Sensor().ExpectConnections(s.T(), client.id, 5*time.Second, clientConnection)
+	s.Sensor().ExpectConnections(s.T(), server.id, 5*time.Second, serverConnection)
+}
+
+func (s *UdpNetworkFlow) TestMultipleDestinations() {
+	image := config.Images().QaImageByKey("qa-udp")
+
+	servers := make([]containerData, CONTAINER_COUNT)
+	clientConnections := make([]types.NetworkInfo, CONTAINER_COUNT)
+	for i := 0; i < CONTAINER_COUNT; i++ {
+		name := fmt.Sprintf("%s-%d", UDP_SERVER, i)
+		port := uint16(9000 + i)
+		servers[i] = s.runServer(config.ContainerStartConfig{
+			Name:    name,
+			Image:   image,
+			Command: newServerCmd("recvfrom", port),
+		}, port)
+		log.Info("Server: %s\n", servers[i].String())
+
+		// Load the client connection collector has to send for this server.
+		clientConnections[i] = types.NetworkInfo{
+			LocalAddress:   "",
+			RemoteAddress:  fmt.Sprintf("%s:%d", servers[i].ip, servers[i].port),
+			Role:           "ROLE_CLIENT",
+			SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+			CloseTimestamp: types.NilTimestamp,
+		}
+	}
+
+	// We give a big period here to ensure the syscall happens just once
+	// Due to an implementation restriction, the total number of messages
+	// sent must be less than 32.
+	client := s.runClient(config.ContainerStartConfig{
+		Name:       UDP_CLIENT,
+		Image:      image,
+		Command:    newClientCmd("sendmmsg", "300", "8", servers...),
+		Entrypoint: []string{"udp-client"},
+	})
+	log.Info("Client: %s\n", client.String())
+
+	for _, server := range servers {
+		serverConnection := types.NetworkInfo{
+			LocalAddress:   fmt.Sprintf(":%d", server.port),
+			RemoteAddress:  client.ip,
+			Role:           "ROLE_SERVER",
+			SocketFamily:   "SOCKET_FAMILY_UNKNOWN",
+			CloseTimestamp: types.NilTimestamp,
+		}
+		s.Sensor().ExpectConnections(s.T(), server.id, 5*time.Second, serverConnection)
```

Comment:
[nit] it might be more performant to construct the list of all server connections and then Expect all of them, and avoids a worst-case sleep for `5 * len(servers)` seconds

Reply:
The problem is that the test uses 3 separate servers, we need to check them separately, unless we put together a `MultiExpectConnections` method or something that can take a `map[string]types.NetworkInfo` or similar so that we can state what NetworkInfos are expected on which containers.

Reply:
Ahh I see that I am blind, my bad. That totally makes sense!

Reply:
Oh, don't worry, I made this comment to myself like 3 times during development 🤣